schemachangerccl: move generated BACKUP tests into subpackage

rafiss opened the pull request:
This patch moves tests into a subpackage since we are starting to
see schemachangerccl timeout due to having too many tests in the
package.

fixes https://github.com/cockroachdb/cockroach/issues/146865
fixes https://github.com/cockroachdb/cockroach/issues/146985
Release note: None

rafiss committed:
schemachangerccl: move generated BACKUP tests into subpackage

This patch moves tests into a subpackage since we are starting to
see schemachangerccl timeout due to having too many tests in the
package.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146885)
<!-- Reviewable:end -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146885#-:-OQPqEgmA-1IkxTw6D5H:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

rafiss commented:
bors r+

rafiss commented:
bors r+

rafiss commented:
bors retry

rafiss committed:
schemachangerccl: move more BACKUP tests to subpackage

This commit follows up to the previous commit by also moving all backup
tests that rely on mulriregion features to a subpackage.

Release note: None

rafiss commented:
bors r+
